src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorByteOffsetFirstFaulting.template

```diff
@@ -0,0 +1,764 @@
+ * script in tests\src\JIT\HardwareIntrinsics\Arm\Shared. In order to make    *
+ * changes, please update the corresponding template and run according to the *
+ * directions listed in the file.                                             *
+ ******************************************************************************/
+
+using System;
+using System.Buffers;
+using System.Numerics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+namespace JIT.HardwareIntrinsics.Arm
+{
+    public static partial class Program
+    {
+        [Fact]
+        public static void {TestName}()
+        {
+            var test = new SveGatherVectorIndices__{TestName}();
+
+            if (test.IsSupported)
+            {
+                // Validates basic functionality works, using Unsafe.Read
+                test.RunBasicScenario_UnsafeRead();
+
+                // Validates basic functionality works
+                test.RunBasicScenario_Load();
+
+                // Validates basic functionality of first-faulting behavior
+                test.RunBasicScenario_LoadFirstFaulting();
+
+                // Validates fully masked out load works.
+                test.RunBasicScenario_FalseMask();
+
+                // Validates fully masked out load with invalid address works.
+                test.RunBasicScenario_NonFaulting();
+
+                // Validates calling via reflection works, using Unsafe.Read
+                test.RunReflectionScenario_UnsafeRead();
+
+                // Validates passing a local works, using Unsafe.Read
+                test.RunLclVarScenario_UnsafeRead();
+
+                // Validates passing an instance member of a class works
+                test.RunClassFldScenario();
+
+                // Validates passing the field of a local struct works
+                test.RunStructLclFldScenario();
+
+                // Validates passing an instance member of a struct works
+                test.RunStructFldScenario();
+
+                // Validates using inside ConditionalSelect with value falseValue
+                // Currently, using this operation in ConditionalSelect() gives incorrect result
+                // when falseReg == targetReg because this instruction uses Pg/Z to update the targetReg
+                // instead of Pg/M to merge it. As such, the value of falseReg is lost. Ideally, such
+                // instructions should be marked similar to RMW (a different flag name) to make sure that
+                // we do not assign falseReg/targetReg same. Then, we would do something like this:
+                //
+                // ldnf1sh target, pg/z, [x0]
+                // sel mask, target, target, falseReg
+                //
+                // This needs more careful thinking, so disabling it for now.
+                // test.ConditionalSelect_FalseOp();
```

Comment:
Will you send a follow-up PR to uncomment this once [https://github.com/dotnet/runtime/pull/105737](https://github.com/dotnet/runtime/pull/105737) is merged or can you merge [https://github.com/dotnet/runtime/pull/105737](https://github.com/dotnet/runtime/pull/105737) first and uncomment this one?

Reply:
This scenario is also disabled in [`SveGatherVectorFirstFaultingIndices.template`](https://github.com/dotnet/runtime/blob/adb1fee95cac68b1a64561c9f414966964fee408/src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorFirstFaultingIndices.template#L73) and [`SveGatherVectorFirstFaultingVectorBases.template`](https://github.com/dotnet/runtime/blob/adb1fee95cac68b1a64561c9f414966964fee408/src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorFirstFaultingVectorBases.template#L73), so maybe we can re-enable it across all templates as a follow-up to #105737?

Reply:
Created an issue for this: #106279